added blog framework

juskuz opened the pull request.

juskuz committed:
Blog framework added. Updated year in footer.
Updated description of adding content.

juskuz commented:
It's https://github.com/pyladies-poland/pyladies-poland.github.io-src/issues/10

juskuz commented:
Blog content should be written in content/blog. 
If this directory is empty there's no "BLOG" section on website.

@aniawsz 
What is "tag for blog articles"?

aniawsz commented on themes/pyladiestheme/templates/article.html:
```diff
@@ -22,7 +22,7 @@ <h2 class="title">{{ article.title }}</h2>
             <div class="row">
                 <div class="col-sm-12">
-                  <a href="/events.html"><h4 class="back">WSZYSTKIE WYDARZENIA</h4></a>
+                  <a href="/events.html"><h4 class="back">222WSZYSTKIE WYDARZENIA</h4></a>
```
I guess that's accidental :)

aniawsz commented on themes/pyladiestheme/templates/base.html:
```diff
@@ -48,6 +48,7 @@
             <!-- <li><a data-scroll href="/#resources">Materiały</a></li> -->
             <li><a data-scroll href="/#coc">Kodeks</a></li>
             <li><a data-scroll href="/#partners">Partnerzy</a></li>
+            <li><a href="/#blog">Blog</a></li>
```
Isn't the class `data-scroll` needed here? What is it needed for then? ( I really don't know :P )

aniawsz commented on themes/pyladiestheme/templates/article.html:
```diff
@@ -22,7 +22,7 @@ <h2 class="title">{{ article.title }}</h2>
             <div class="row">
                 <div class="col-sm-12">
-                  <a href="/events.html"><h4 class="back">222WSZYSTKIE WYDARZENIA</h4></a>
+                  <a href="/blog/"><h4 class="back">222WSZYSTKIE WYDARZENIA</h4></a>
```
oh I guess it was just for testing? :)

juskuz replied:
Yep, that was only for testing :)

aniawsz reviewed (commented):
I left some comments and questions, happy to chat when something's not cleat!

When by `tag` I actually meant `category`! Sorry 😬 
There's a plugin for grouping content items into categories (https://github.com/pyladies-poland/pyladies-poland.github.io-src/blob/master/plugins/categories_dict.py). Then it's easy to use it in templates. Sorry for causing the confusion!

juskuz commented:
@aniawsz 
Thanks for your review. I'll be working on it next week ;)

magul commented:
@aniawsz We just spoke with @juskuz and can we implement tag/categories functionality in another pull request?

BTW when You write tag/categories, You mean that we want to categorize blog posts?

juskuz commented:
I pushed two example blog posts (to see how it works). Before deploying you need to remove it :)
If there is no files in 'content/blog' there'll be no visible button "BLOG" or section "BLOG" on website.

aniawsz commented on themes/pyladiestheme/templates/base.html:
```diff
@@ -90,7 +92,7 @@
         <div class="row">
           <div class="col-sm-3">
-            <p class="copyright">2017 &copy; PyLadies Poland</p>
+            <p class="copyright">2018 &copy; PyLadies Poland</p>
```
🎉 👍

aniawsz commented on SETUP.md:
```diff
@@ -5,6 +5,7 @@ git clone https://github.com/pyladies-poland/pyladies-poland.github.io-src
 2. Aby dodać:
   * osobę, dodaj plik rst do folderu content/people
   * wydarzenie, dodaj plik rst do folderu content/events
+  * post na bloga, dodaj plik rst do folderu content/blog. Pamiętaj o dodaniu ':template: blog_post' oraz ':category: blog'.
```
👌

aniawsz reviewed (approved):
Awesome job! 👏
Just two minor things:
1) I'm a bit worried that we're breaking the consistency of having alternating backgrounds on the main page (white / dark), but have no ideas on how to fix that; since the Partners section needs to be white (for logos to look better). I'd say let's leave it as it is for now and hope for another section to come up soon :) If you have any other ideas, let's discuss!
2) It would be nice to clean up the commit history a bit before merging (squash commits introducing and later deleting the same lines, rebase on top of the current master to avoid in-between merges).

Please let me know if you need help with any of that. Great job! 🎉

juskuz commented:
@aniawsz 
Thanks for your review. 

I've made little changes (I removed example articles). And then I've cleaned up commit history. Now there is only one commit :) It looks much better.

And if we want dark website background, then maybe logos should be in gray scale (with white background around them)? And they will be changing into colorful when moving the mouse pointer onto them? But I'm not sure it'll look fine (I haven't seen dark version).
